extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -536,28 +559,77 @@ export class AmpStoryEmbeddedComponent {
           const targetRect = target./*OK*/getBoundingClientRect();
           const pageRect = this.componentPage_./*OK*/getBoundingClientRect();
 
-          const centeredTop = pageRect.height / 2 - targetRect.height / 2;
-          const centeredLeft = pageRect.width / 2 - targetRect.width / 2;
-
-          // Only account for offset from target to page borders. Since in
-          // desktop mode page is not at the borders of viewport.
-          const leftOffset = targetRect.left - pageRect.left;
-          const topOffset = targetRect.top - pageRect.top;
-
-          state.translateY = centeredTop - topOffset;
-          state.translateX = leftOffset - centeredLeft;
+          this.scaleFactor_ =
```

Comment:
I'm not sure I get what's going on over the next ~12 lines; not saying that it's wrong, but I'm just unable to follow.  Can you explain (in GitHub, and then subsequently in code-level comments)?

Reply:
Why do we need to get the fullscreen width from the scaleFactor when we could use the `pageRect` coordinates?

Reply:
I know you're working on this, but now that you're going to handle resize events better, it'd be cool if we don't have to compute everything twice :))

Reply:
Done.